Dashboard: Adds folder name and link to the dashboard overview on the homepage

michelengelen opened the pull request:
Adds the `folderTitle` of the Dashboard to the Dashboard overview on the Grafana home.

fixes #26869

Commit:
Chore: Added foldername to the dashboard overview on the startpage

## Dashlist

### added
- new `<span>` containing `dash.folderTitle`
- styling for the newly created `<span>`

### changed
- added sass-nesting to classes in `_panel_dashlist.scss`

closes #26869

hugohaggmark commented on public/app/plugins/panel/dashlist/module.html:
```diff
@@ -9,6 +9,9 @@ <h6 class="dashlist-section-header" ng-show="ctrl.panel.headings">
           <span class="dashlist-title">
             {{dash.title}}
           </span>
+          <span class="dashlist-folder">
+            ({{dash.folderTitle}})
+          </span>
```
Maybe we should ignore empty folder names as it looks kind of weird:
![image](https://user-images.githubusercontent.com/562238/90094209-9ae22480-dd2d-11ea-9629-e7312bd3e7ae.png)

michelengelen replied:
Hi @hugohaggmark 

I have added a `ng-if` directive to the span. 
Now it does not display the empty folder anymore, when no `folderTitle` property is present.

It should be fine now.

WDYT?

hugohaggmark reviewed (commented):
Thank you for contributing @michelengelen but I left a small comment for you to consider. Also what do you think about this feature @torkelo ?

Commit:
Chore: Added foldername to the dashboard overview on the startpage

## Dashlist

### changed
- added `ng-if` directive to the dashlist-folder span

dprokop commented:
@hugohaggmark @torkelo I think (feature-wise) this is OK. I understand the need here in a case where there are dashboards with the same names across different folders. One thing to consider is to consult with @jessover9000 if she doesn't have other ideas how to solve this issue.

torkelo commented:
Think this is fine, but long term we should use the same card as search results I think (but need to rewrite this panel to react first)

jessover9000 commented:
I think it's a good idea to show the folder names! However, from a UX perspective putting the folder name in brackets doesn't seem like an established pattern to indicate a file location. If I looked at that view without explanation, I would probably wonder what that text in the brackets means – I'm not sure whether I could recognize it's a folder name, especially if I don't know what folders there are in the system. So maybe instead of showing the brackets, you could just add a little folder icon like I quickly mocked up in the attached screenshot?

![image](https://user-images.githubusercontent.com/53754871/90117895-cda11280-dd57-11ea-8a9f-a00cc822ff83.png)

michelengelen commented:
The solution with the folder icon is way better ... I agree! I will update the PR in the evening (I am on vacation atm 🌴)

@torkelo if you need help migrating the panel to react I would be glad to help in this regard as well. Just add me to the issue when it is ready!

Commit:
Chore: Added foldername to the dashboard overview on the startpage

## Dashlist

### changed
- added folder-icon and removed brackets to indicate `folderTitle`
- changed font-size of folderTitle-`span` and added left margin
